### PR TITLE
Ability to delete a project.

### PR DESCRIPTION
**Pull request by prodex**
issue #46 

Done:
- Mark project for deletion. 
Removal is possible only from draft. The project is permanently deleted.
- Hide deleted projects in lists.
- Queue to delete project resources such as images, bookmarks and etc.
- API.

PS: I wanted as little code as possible) But I had to add logic to handle the deleted status.

### CODE REVIEW

File: controllers/ProjectController.php [hunk 1]
```diff
@@ -52,7 +52,8 @@ public function behaviors()
                 'class' => VerbFilter::class,
                 'actions' => [
                     'publish' => ['post'],
-                    'draft' => ['post']
+                    'draft' => ['post'],
+                    'delete' => ['post'],
```
Comment (samdark) [inline controllers/ProjectController.php]: Shouldn't `delete` action be mapped to `DELETE` HTTP method?

Comment (prodex) [reply controllers/ProjectController.php]: Maybe you messed up with the "delete" method in the API? In app\controllers\ProjectController "delete" method should not be called through GET.

File: modules/api1/controllers/ProjectController.php [hunk 2]
```diff
@@ -102,6 +106,27 @@ public function actionVote($id)
...
+     * @throws UnauthorizedHttpException
+     */
+    public function actionDelete($id)
+    {
+        $user = $this->getCurrentUser();
+        if (!$user) {
+            throw new UnauthorizedHttpException('User should be authorized in order to manage project.');
+        }
+        
+        $project = $this->findProject($id, $user);
```
Comment (samdark) [inline modules/api1/controllers/ProjectController.php]: I think the following check is missing:

```php
if (!UserPermissions::canManageProject($project)) {
    throw new ForbiddenHttpException(Yii::t('project', 'You can not delete this project.'));
}
```

File: controllers/ProjectController.php (hunk 1 above)
Comment (samdark) [reply controllers/ProjectController.php]: Yes.

File: modules/api1/controllers/ProjectController.php (hunk 2 above)
Comment (prodex) [reply modules/api1/controllers/ProjectController.php]: I add the condition. But still as an additional check, the code below only skips the owner.
````
protected function findProject($projectId, User $user)
    {
        /** @var Project $project */
        $project = Project::find()
            ->where(['id' => $projectId])
            ->hasUser($user)
            ->available()
            ->one();

        if ($project) {
            return $project;
        }
        
        throw new NotFoundHttpException("The requested project does not exist.");
    }
```

**Comment by prodex** [comment]
Done.

File: modules/api1/controllers/ProjectController.php (hunk 2 above)
Comment (samdark) [reply modules/api1/controllers/ProjectController.php]: Ah. Alright.

**Comment by samdark** [comment]
Will check it once again and deploy soon.

**Comment by prodex** [comment]
ok)

I have a question: why relations `project` - `user` done as many to many?

**Comment by samdark** [comment]
Well, a team could do a project where multiple users want to be displayed and be able to manage the project. Originally it was planned to be able to add users to a project.

**Comment by samdark** [comment]
Merged. Thanks!

**Pull request merged**